### PR TITLE
Update dependencies

### LINKED ISSUE #167: Address dependency management issues
https://github.com/ciena-frost/ember-frost-sort/issues/76 **MUST** be completed before this issue.

    * remove from blueprints and make dependency
        * ember-frost-core
        * ember-frost-sort
        * ember-math-helpers
        * smoke-and-mirrors

    * remove from devDependency and make dependency
        * ember-prop-types
        * ember-spread

    * update ember-frost-sort
    * update ember-prop-types to ^5.0.1
    * update ember-spread to ^3.0.2

    * check whether these are specifically needed or not and remove if not, otherwise move to dependencies
        * ember-computed-decorators
        * ember-concurrency
        * ember-elsewhere
        * ember-math-helpers
        * ember-truth-helpers
        * smoke-and-mirrors

### PR DESCRIPTION
**Pull request by juwara0**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [X] #major# - incompatible API change

Closes: https://github.com/ciena-frost/ember-frost-list/issues/167

# CHANGELOG
* **Added** ignore the linting of the `CHANGELOG.md`
* **Added** ignoring of `package-lock` until we are ready to move to node 8
* **Removed** useLintTree ember-cli-mocha configuration from `ember-cli-build.js`
* **Removed** `.remarkrc` file since it is now provided by `ember-test-utils`
* **Updated** `ember-frost-test` to `^4.0.0`
* **Updated** `ember-cli-chai` to `0.4.3`
* **Updated** `ember-cli-mocha` to `0.14.4`
* **Updated** `ember-sinon` to `^0.7.0`
* **Updated** `ember-test-utils` to `^8.1.0`
* **Updated** `sinon-chai` @ `^2.14.0`
* **Updated** `ember-cli-code-coverage` to `0.3.12`
* **Updated** `ember-cli-frost-blueprints` to `^5.0.1`
* **Updated** `ember-cli-notifications` to `^4.2.1`
* **Updated** pin `ember-code-snippet` to `1.7.0`
* **Updated** `ember-frost-sort` to `^10.0.0`
* **Removed** unused `ember-async-image` package
* **Removed** unused `pull-report` package
* **Added** `broccoli-funnel` @ `^2.0.1`
* **Added** `broccoli-merge-trees` @ `^2.0.0`
* **Updated** `ember-cli-sass` to `7.1.1`
* **Updated** `ember-frost-core` to `^5.1.1`
* **Updated** `ember-hook` to `1.4.2`
* **Updated** `ember-prop-types` to `^6.0.1`
* **Updated** `ember-truth-helpers to `^1.3.0`
* **Updated** `ua-parser-js` to `^0.7.17`
* **Removed** static copy of `ua-parser-js.js` located in `vendor/`
* **Added** a vendor shim to enable import of `ua-parser-js` npm package
* **Updated** `index.js` file to add `ua-parser-js` to this add-on's build tree
* **Updated** usage of UAParser to use new importable module instead of global
* **Updated** move code coverage config file to tests/dummy/config/
* **Removed** unused `animation-frame` bower package
* **Updated** `Faker` and `jquery-mockjax` bower packages to be devDependencies

### CODE REVIEW

File: vendor/shims/ua-parser-js.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    'use strict';
+
+    return {
+      'default': self['UAParser'],
+      __esModule: true,
+    };
+  }
+
+  define('ua-parser-js', [], vendorModule);
+})();
```
Comment (notmessenger) [inline vendor/shims/ua-parser-js.js]: Do we want a semi-colon at the end of this?  Or any of the lines for that matter.

As an aside, do we need to open an issue for our linter to lint these files as well?

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-list/pull-request/171/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-list)

File: vendor/shims/ua-parser-js.js (hunk 1 above)
Comment (notmessenger) [reply vendor/shims/ua-parser-js.js]: I opened https://github.com/ciena-blueplanet/ember-test-utils/issues/107

Comment (juwara0) [reply vendor/shims/ua-parser-js.js]: This has been updated to comply with our linting standards.

**Pull request merged**